core-js version conflicts when mixing @nrwl/angular and @nrwl/react into the same workspace

_[Please make sure you have read the submission guidelines before posting an issue](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#-submitting-issue)_

# Prerequisites

Please answer the following questions for yourself before submitting an issue.
**YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version: `"@nrwl/workspace": "8.8.0",`
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I'm reporting the issue to the correct repository (not related to Angular, AngularCLI or any dependency)

## Expected Behavior

Nx can work in a monorepo with both `@nrwl/angular` and `@nrwl/react`.
Hopefully this is a valid use case for Nx.

## Current Behavior

Adding a react app into an existing workspace (there's already an angular app), the react app cannot be started

What is the current behavior?

As above.

## Failure Information (for bugs)

Please help provide information about the failure if this is a bug. If it is not a bug, please remove the rest of this template.

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. create workspace:
`yarn create nx-workspace --preset=empty myworkspace`

2. generate an angular app first:
`yarn add --dev @nrwl/angular && yarn ng g @nrwl/angular:app myapp1`
this app can be started when running `yarn start`

3. generate another react app:
`yarn add -D @nrwl/react && yarn nx g @nrwl/react:app myapp2`
this react app cannot be started when running `yarn start myapp2`, there were a lot of errors about `core-js` thrown in `polyfills.ts`:
```
ERROR in ./polyfills.ts
Module not found: Error: Can't resolve 'core-js/modules/es.array-buffer.constructor'
```

checked that it's actually from:
```js
import 'core-js/stable'; // this is actually since v3 only
```
and the version for `core-js` in package.json: `"core-js": "^2.5.4",`

### Context

Please provide any relevant information about your setup:

- version of Nx used (_Please run `nx report` at the root of your workspace and copy the results here if you are using Nx 8.6.1 or greater_) `8.8.0`
- 3rd-party libraries and their versions
- and most importantly - a use-case that fails

### Failure Logs

As above

## Other

Any other relevant information that will help us help you.

--- a/e2e/utils.ts
+++ b/e2e/utils.ts
@@ -214,12 +214,7 @@ export function copyMissingPackages(): void {
     '@testing-library',
 
     // For testing webpack config with babel-loader
-    '@babel/core',
-    '@babel/preset-env',
-    '@babel/preset-react',
-    '@babel/preset-typescript',
-    '@babel/plugin-proposal-class-properties',
-    '@babel/plugin-proposal-decorators',
+    '@babel',
     'babel-loader',
     'babel-plugin-macros',
     'eslint-plugin-import',

--- a/packages/web/src/utils/config.ts
+++ b/packages/web/src/utils/config.ts
@@ -1,6 +1,6 @@
 import * as webpack from 'webpack';
 import { Configuration, ProgressPlugin, Stats } from 'webpack';
-import { dirname } from 'path';
+import { dirname, resolve } from 'path';
 import { LicenseWebpackPlugin } from 'license-webpack-plugin';
 import * as CopyWebpackPlugin from 'copy-webpack-plugin';
 import * as TerserWebpackPlugin from 'terser-webpack-plugin';
@@ -63,6 +63,9 @@ export function getBaseWebpackPartial(
           mainFields
         })
       ],
+      // Search closest node_modules first, and then fallback to to default node module resolution scheme.
+      // This ensures we are pulling the correct versions of dependencies, such as `core-js`.
+      modules: [resolve(__dirname, '..', '..', 'node_modules'), 'node_modules'],
       mainFields
     },
     performance: {

